Firestore/core/src/firebase/firestore/remote/grpc_connection.mm

```diff
@@ -52,6 +54,37 @@
   return view.data() ? std::string{view.data(), view.size()} : std::string{};
 }
 
+std::string LoadCertificate(const Path& path) {
+  std::ifstream certificate_file{path.native_value()};
+  HARD_ASSERT(certificate_file.good(),
+              StringFormat("Unable to open root certificates at file path %s",
+                           path.ToUtf8String())
+                  .c_str());
+
+  std::stringstream buffer;
+  buffer << certificate_file.rdbuf();
+  std::string certificate = buffer.str();
+  // Certificate is in UTF-8 and may contain non-ASCII characters in comments,
+  // which may not be supported by gRPC. Replacing them is faster than removing
```

Comment:
Is this necessary? What error do you get if you don't do this?

I ask, because this is weird. It would seem that all grpc needs to do is *not* read the comments and it should be ok. (Going one step further, grpc presumably does very little or maybe no interpretation of this file, since the ssl part would be implemented by boring/open ssl.) Glancing briefly at the grpc code, they don't seem to take any similar action when reading their own roots.pem file...? (But I haven't gone through it in detail.)

Reply:
Removed, thanks for calling it out. I did this because that's what gRPC Objective-C client [does](https://github.com/grpc/grpc/blob/master/src/objective-c/GRPCClient/private/GRPCHost.m#L161-L162). However, the actual problem there is that originally `NSString` tried to load the string as ASCII, encountered non-ASCII characters, and produced `nil`. I'm not sure why gRPC uses lossy conversion rather than just loading the string as UTF-8, which it is. Anyway, this doesn't apply to C++. (I also found a bunch of historical OpenSSL bugs with handling unicode in certificates, but they look fixed and likely were not applicable to our case anyway). Thanks for pushing me to investigate.